api/src/test/java/org/apache/iceberg/expressions/TestStringLiteralConversions.java

```diff
@@ -101,7 +101,7 @@ public void testStringToTimestampLiteral() {
 
     // Timestamp with explicit UTC offset, +00:00
     Literal<CharSequence> timestampStr = Literal.of("2017-08-18T14:21:01.919+00:00");
-    Literal<Long> timestamp = timestampStr.to(Types.TimestampType.withZone());
+    Literal<Long> timestamp = timestampStr.to(Types.TimestampType.microsWithZone());
```

Comment:
Don't we need similar tests for nanoseconds?

Reply:
Will do.

Reply:
Done.